katdal/sensordata.py

```diff
@@ -454,10 +453,12 @@ def remove_duplicates_and_failures(sensor):
             logger.debug("At %s, sensor %r has statuses of %r and %r - "
                          "keeping last one", katpoint.Timestamp(x[ind]).local(),
                          sensor.name, z[ind], z[replacement][ind])
-    # Remove entries where 'status' equals 'failure', if 'status' is present
+    # Remove entries where 'status' implies invalid values, if 'status' is present
     if z is not None:
         # Explicitly cast status to string type, as k7_augment produced sensors with integer statuses
-        unique_ind = unique_ind[z[unique_ind].astype('|S7') != 'failure']
+        status = z[unique_ind].astype('|S7')
+        has_valid_value = np.array([['nominal'], ['warn'], ['error']])
+        unique_ind = unique_ind[(status == has_valid_value).any(axis=0)]
```

Comment:
This seems a little complicated. How about `unique_ind[(status == 'nominal') | (status == 'warn') | (status == 'error')]`?